pkgs/development/python-modules/online-judge-tools/api-client.nix

```
@@ -0,0 +1,41 @@
+{ appdirs, beautifulsoup4, buildPythonPackage, colorlog, fetchFromGitHub, git
+, jsonschema, lib, lxml, markdown, python, requests, substituteAll, toml }:
+
+let
+  pythonEnv =
+    python.withPackages (ps: with ps; [ colorlog jinja2 markdown toml ]);
+in buildPythonPackage rec {
```

```suggestion
buildPythonPackage rec {
```

I am not sure if we are allowed to do this in pythonPackages but my gut feeling says no.

---

As we discussed https://github.com/NixOS/nixpkgs/pull/146267#discussion_r750499985, this is necessary for online-judge-tools to internally download [library-checker-problems](https://github.com/yosupo06/library-checker-problems) and invoke `generate.py` in it.

Is there a better way?
I already tried `propagatedBuildInputs` and it didn't work.

---

I considered I should make a derivation for library-checker-problems and contain it in online-judge-tools.
But library-checker-problems is more contents than a tool, and is often updated. So I don't want to pin its revision.

---

If nothing of this is propagated this should not be a problem. Please add a comment that other people know that.